src/index.ts

```diff
@@ -90,13 +100,34 @@ class Compiler {
   }
 
   // eg. "#/definitions/diskDevice" => ["definitions", "diskDevice"]
-  private resolveType(path: string): TsType.TsType {
-    if (path[0] !== '#')
-      throw new Error('reference must start with #')
-    if (path === '#' || path === '#/')
+  // only called in case of a $ref type
+  private resolveType(refPath: string, propName: string): TsType.TsTypeBase {
+    if (refPath === '#' || refPath === '#/'){
       return TsType.Interface.reference(this.id)
-    const parts = path.slice(2).split('/')
+    }
+
+    if (refPath[0] !== '#'){
+      let id: string
+      let fullPath = resolve(join(this.filePath.dir, refPath))
+      let file = readFileSync(fullPath)
```

Comment:
Let's do some error handling here - what if this fails? A ReferenceError would be helpful if this is an http:// url, or points to an invalid file.